torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -491,28 +493,33 @@ class NCCLComm {
         ncclComm_);
 
     void* handle = registeredSegmentHandles_[ptr];
+    // Use getNcclComm to make sure comm is ready before calling nccl APIs
+    auto comm = getNcclComm();
     C10D_NCCL_CHECK(
-        ncclCommDeregister(ncclComm_, handle),
+        ncclCommDeregister(comm, handle),
         c10::str(
             "Failed to deregister segment handle ",
             handle,
             ", with ptr ",
             ptr,
             " on ncclComm_ ",
-            ncclComm_));
+            comm));
     registeredSegmentHandles_.erase(ptr);
     return ncclSuccess;
 #else
     return ncclInvalidUsage;
 #endif
   }
 
+  std::string repr() const {
```

Comment:
what does repr mean here? use addr()?

Reply:
Yeah just ncclComm_ address, because ProcessGroupNCCL was printing it.